fdbcli/LockCommand.actor.cpp

```diff
@@ -0,0 +1,120 @@
+/*
+ * LockCommand.actor.cpp
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "fdbcli/fdbcli.actor.h"
+
+#include "fdbclient/FDBOptions.g.h"
+#include "fdbclient/IClientApi.h"
+#include "fdbclient/Knobs.h"
+#include "fdbclient/Schemas.h"
+
+#include "flow/Arena.h"
+#include "flow/FastRef.h"
+#include "flow/ThreadHelper.actor.h"
+#include "flow/actorcompiler.h" // This must be the last #include.
+
+namespace {
+
+ACTOR Future<Void> lockDatabase(Reference<IDatabase> db, UID id) {
+	state Reference<ITransaction> tr = db->createTransaction();
+	loop {
+		tr->setOption(FDBTransactionOptions::SPECIAL_KEY_SPACE_ENABLE_WRITES);
+		try {
+			tr->set(fdb_cli::lockSpecialKey, id.toString());
+			wait(safeThreadFutureToFuture(tr->commit()));
+			return Void();
+		} catch (Error& e) {
+			if (e.code() == error_code_database_locked)
+				throw e;
+			state Error err(e);
+			if (e.code() == error_code_special_keys_api_failure) {
+				std::string errorMsgStr = wait(fdb_cli::getSpecialKeysFailureErrorMessage(tr));
+				if (errorMsgStr == "Database has already been locked")
```

Comment:
This seems a little dangerous to me, if this string ever changes it would be easy to forget to update it everywhere. Is there code coverage here? Maybe you could add a `TEST`, or maybe an `fdb_cli` function that returns true/false if the database is locked based on the state of the transaction object.

Reply:
Nice catch!
I somehow realized I do not need `special_keys_api_failure` as I can throw `database_locked` directly.
And also the error handling here is messed up.
Let me update them.

Reply:
Done. Get rid of this message, throw `database_locked` directly in the special keys' implementation.